pkg/controller/backup_controller.go

```diff
@@ -281,6 +286,53 @@ func (c *backupController) getLocationAndValidate(itm *api.Backup, defaultBackup
 	return backupLocation, validationErrors
 }
 
+// defaultAndValidateSnapshotLocations ensures:
+// - each location name in Spec VolumeSnapshotLocation exists as a location
+// - exactly 1 location per existing or default provider
+// - a given default provider's location name is added to the Spec VolumeSnapshotLocation if it does not exist as a VSL
+func (c *backupController) defaultAndValidateSnapshotLocations(itm *api.Backup, defaultLocations map[string]string) []string {
```

Comment:
Should `defaultLocations map[string]string` be `defaultLocations []map[string]string` instead?

Reply:
Need to know this ^ before I finish fixing the server cmd flag bug.

Reply:
no - should be a map of provider->VSL name